app/forms/contact_details_form.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+class ContactDetailsForm
+  include ActiveModel::Model
+  include ActiveModel::AttributeAssignment
+  include ActiveModel::Validations::Callbacks
+
+  MANDATORY_UK_ADDRESS_FIELDS = %i[
+    address_line_one
+    town_city
+    postcode
+  ].freeze
+
+  FIELDS = %i[
+    locale_code
+    address_line_two
+    international_address
+    email
+  ].concat(MANDATORY_UK_ADDRESS_FIELDS).freeze
+
+  delegate :id, :persisted?, to: :trainee
+
+  attr_accessor(*FIELDS, :fields, :trainee)
+
+  before_validation :sanitise_email
+
+  validates :email, presence: true
+  validates :locale_code, presence: true
+  validate :uk_address_valid, if: -> { uk? }
+  validate :international_address_valid, if: -> { !uk? }
```

Comment:
Currently, `!uk?` will return `true` if `fields[:locale_code]` is `nil`. This means this validation runs before we've even selected a locale code.